src/groupeddataframe/utils.jl

```diff
@@ -323,45 +417,123 @@ function row_group_slots(cols::NTuple{N, AbstractVector},
             end
             refmap
         end
-        @inbounds @spawn_for_chunks 1_000_000 for i in eachindex(groups)
-            local refs_i
-            let i=i # Workaround for julia#15276
-                refs_i = map(c -> c[i], refarrays)
+        if use_threading
+            Threads.@threads for i in eachindex(groups)
+                @inbounds begin
+                    seeni = seen_vec[Threads.threadid()]
```

Comment:
Instead of duplicating the code in the other branch, I think you could always take this branch even when `!use_threading`, since `seen_vec[1] === seen` already.

Reply:
no - this branch will ALWAYS spin `Threads.nthreads()` processes, which we must avoid.

Reply:
Ah right. Could we define a simple `@threadsif use_threading for...` macro that would duplicate the code internally?

Reply:
I tried to do it, but failed. Essentially you need to conditionally inject `Threads.@threads` in the code, but macro expander then failed to properly process it. But if you know how to do it then of course it would be better.

Reply:
Something like this?
```julia
macro threadsif(cond, ex)
    quote
        if $cond
            Threads.@threads $ex
        else
            $ex
        end
    end
end
```

Reply:
This is incorrect as it is incorrect hygene AFAICT. And when you fix hygene:
```
macro threadsif(cond, ex)
    quote
        if $esc(cond)
            Threads.@threads $esc(ex)
        else
            $esc(ex)
        end
    end
end
```
then it is not processed correctly.

Reply:
Too bad. Also I've realized that when `use_threading` is false, `threadid()` is not necessarily 1, so `seen_vec[Threads.threadid()]` is not necessarily valid -- we would have to allocate a vector filled with `seen`. Yet all this code duplication doesn't feel right. :-/

Reply:
we could fill all `seen` with the same vector, but still I am not sure how to cleanly write a macro that conditionally prepends `Threads.@threads`.

Reply:
resolved with migration to `@similar`